drivers/spi/spi_mcux_flexio.c

```diff
@@ -0,0 +1,431 @@
+		} else if (word_size == 16) {
+			data->transfer_flags = kFLEXIO_SPI_16bitLsb;
+		} else {
+			data->transfer_flags = kFLEXIO_SPI_32bitLsb;
+		}
+	} else {
+		if (word_size == 8) {
+			data->transfer_flags = kFLEXIO_SPI_8bitMsb;
+		} else if (word_size == 16) {
+			data->transfer_flags = kFLEXIO_SPI_16bitMsb;
+		} else {
+			data->transfer_flags = kFLEXIO_SPI_32bitMsb;
+		}
+	}
+
+	if (mcux_flexio_get_rate(config->flexio_dev, &clock_freq)) {
+		return -EINVAL;
+	}
+
+	master_config.phase =
+		(SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA)
+		? kFLEXIO_SPI_ClockPhaseSecondEdge
+		: kFLEXIO_SPI_ClockPhaseFirstEdge;
+
+	master_config.baudRate_Bps = spi_cfg->frequency;
+	spi_flexio_master_init(config->flexio_spi, &master_config,
+		(SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPOL), clock_freq);
+
+	FLEXIO_SPI_MasterTransferCreateHandle(config->flexio_spi, &data->handle,
+					spi_mcux_master_transfer_callback,
+					data);
+	/* No SetDummyData() for FlexIO_SPI */
+
+	data->ctx.config = spi_cfg;
+
+	return 0;
+}
+
+
+static int transceive(const struct device *dev,
+			const struct spi_config *spi_cfg,
+			const struct spi_buf_set *tx_bufs,
+			const struct spi_buf_set *rx_bufs,
+			bool asynchronous,
+			spi_callback_t cb,
+			void *userdata)
+{
+	const struct spi_mcux_flexio_config *config = dev->config;
+	struct spi_mcux_flexio_data *data = dev->data;
+	int ret;
+
+	spi_context_lock(&data->ctx, asynchronous, cb, userdata, spi_cfg);
+
+	mcux_flexio_lock(config->flexio_dev);
+	ret = spi_mcux_flexio_configure(dev, spi_cfg);
+	mcux_flexio_unlock(config->flexio_dev);
+	if (ret) {
+		goto out;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	spi_context_cs_control(&data->ctx, true);
+
+	mcux_flexio_lock(config->flexio_dev);
+	mcux_flexio_irq_disable(config->flexio_dev);
+
+	spi_mcux_transfer_next_packet(dev);
+
+	mcux_flexio_irq_enable(config->flexio_dev);
+	mcux_flexio_unlock(config->flexio_dev);
+
+	ret = spi_context_wait_for_completion(&data->ctx);
+out:
+	spi_context_release(&data->ctx, ret);
+
+	return ret;
+}
+
+static int spi_mcux_transceive(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, false, NULL, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_mcux_transceive_async(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs,
+				spi_callback_t cb,
+				void *userdata)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, true, cb, userdata);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_mcux_release(const struct device *dev,
+				const struct spi_config *spi_cfg)
+{
+	struct spi_mcux_flexio_data *data = dev->data;
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static int spi_mcux_init(const struct device *dev)
```

Comment:
Do we need to check if the parent FlexIO controller device is ready in this function, since that would indicate if the required shifters/timers for this driver were allocated successfully?

Reply:
@danieldegrasse, There is no need to check the readiness of the parent device because the parent's initialization function does not affect resource management.